isis/src/qisis/objs/Directory/Directory.cpp

```diff
@@ -518,6 +518,15 @@ namespace Isis {
   void Directory::setHistoryContainer(QDockWidget *historyContainer) {
     if (!m_historyTreeWidget) {
       m_historyTreeWidget = new HistoryTreeWidget( project() );
+
```

Comment:
1) what is the TODO?; 2) are you attempting to remove or hide the centralWidget?; 3) maybe...  "The setSizePolicy calls below were tests to get rid of odd behavior with the historyTreeWidget and warningTreeWidget after hiding the central widget. After the two calls immediately below, the warning/history widgets look ok until a view is added. After the view is added they become too big vertically and you can't make them smaller, just larger. With the last two setSizePolicy calls, the history/warning widgets maintain their original size after adding a view, but they can't be made larger.
4) don't understand  "so keeps at sizeHint which is good